### PR TITLE
[SPARK-12974] [ML] [PySpark] Add Python API for spark.ml bisecting k-means

### PR DESCRIPTION
**Pull request by yanboliang**
Add Python API for spark.ml bisecting k-means.

### CODE REVIEW

File: python/pyspark/ml/clustering.py [hunk 1]
```diff
@@ -170,6 +170,129 @@ def getInitSteps(self):
...
+        """
+        __init__(self, featuresCol="features", predictionCol="prediction", maxIter=20, \
+                 seed=None, k=4, minDivisibleClusterSize=1.0)
+        """
+        super(BisectingKMeans, self).__init__()
+        self._java_obj = self._new_java_obj("org.apache.spark.ml.clustering.BisectingKMeans",
+                                            self.uid)
+        self._setDefault(maxIter=20, k=4, minDivisibleClusterSize=1.0)
+        kwargs = self.__init__._input_kwargs
+        self.setParams(**kwargs)
```
Comment (yanboliang) [inline python/pyspark/ml/clustering.py]: After #10216, we do not need to declare param variables in ```__init__``` by hand.

**Comment by yanboliang** [comment]
cc @mengxr @jkbradley

**Comment by mengxr** [comment]
@yanboliang Could you update this PR since #10216 was merged? Thanks!

**Comment by yanboliang** [comment]
@mengxr Actually, I have already updated this PR after #10216 get merged.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**